fluffy/views.py

```diff
@@ -170,82 +171,141 @@ def upload():
+                paste_obj = ctx.enter_context(
+                    HtmlToStore.from_html(
+                        render_template(
+                            'markdown.html',
+                            text=text,
+                            raw_url=app.config['FILE_URL'].format(name=uf.name),
+                        ),
+                    ),
+                )
+                objects.append(paste_obj)
 
-        # Metadata JSON object
-        metadata = {
-            'server_version': FLUFFY_VERSION,
-            'timestamp': time.time(),
-            'upload_type': 'paste',
-            'paste_details': {
-                'urls': {
-                    'html': paste_obj.url,
-                    'raw': uf.url,
-                },
-                'language': {
-                    'title': lang_title,
-                },
-                'num_lines': len(transformed_text.splitlines()),
-                'raw_text': transformed_text,
-            },
-        }
-        metadata_obj = ctx.enter_context(
-            UploadedFile.from_text(
-                json.dumps(metadata, indent=4, sort_keys=True),
-                human_name='metadata.json',
-            ),
-        )
+        metadata = _get_metadata_json(paste_obj, uf, lang_title, transformed_texts_len, transformed_texts)
+        metadata_obj = _get_metadata_obj(metadata, ctx)
         objects.append(metadata_obj)
 
         upload_objects(objects, metadata_obj.url)
 
+        return _generate_output(paste_obj, uf, metadata_obj, lang_title, metadata)
+
+
+@app.route('/upload-history')
+def upload_history():
+    return render_template(
+        'upload-history.html',
+        icon_extensions=ICON_EXTENSIONS,
+    )
+
+
+def _get_raw_text_object(formatted_text, ctx):
+    try:
+        uf = ctx.enter_context(UploadedFile.from_text(formatted_text))
+    except FileTooLargeError as ex:
+        num_bytes, = ex.args
+        return 'Exceeded the max upload size of {} (tried to paste {})'.format(
+            human_size(app.config['MAX_UPLOAD_SIZE']),
+            human_size(num_bytes),
+        ), 413
+    return uf
+
+
+def _get_metadata_json(paste_obj, uf, lang_title, transformed_texts_len, transformed_texts):
+    metadata = {
+        'server_version': FLUFFY_VERSION,
+        'timestamp': time.time(),
+        'upload_type': 'paste',
+        'paste_details': {
+            'urls': {
+                'html': paste_obj.url,
+                'raw': uf.url,
+            },
+            'language': {
+                'title': lang_title,
+            },
+            'num_lines': transformed_texts_len,
+            'raw_text': transformed_texts,
+        },
+    }
+    return metadata
+
+
+def _get_metadata_obj(metadata, ctx):
+    metadata_obj = ctx.enter_context(
+        UploadedFile.from_text(
+            json.dumps(metadata, indent=4, sort_keys=True),
+            human_name='metadata.json',
+        ),
+    )
+    return metadata_obj
```

Comment:
I think this was better inlined, this is very specific to the function calling it and is only used once.